dataflow/src/main/java/org/checkerframework/dataflow/reachingdefinitions/ReachingDefinitionsValue.java

```diff
@@ -0,0 +1,51 @@
+package org.checkerframework.dataflow.reachingdefinitions;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.AbstractValue;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.javacutil.BugInCF;
+
+/**
+ * A reaching definitions (which is represented by a node) wrapper turning node into abstract value.
+ */
+public class ReachingDefinitionsValue implements AbstractValue<ReachingDefinitionsValue> {
+
+    /**
+     * A reaching definition is represented by a node, which can be a {@link
+     * org.checkerframework.dataflow.cfg.node.AssignmentNode}.
+     */
+    protected final AssignmentNode def;
+
+    @Override
+    public ReachingDefinitionsValue leastUpperBound(ReachingDefinitionsValue other) {
+        throw new BugInCF("lub of reachDef get called!");
+    }
+
+    /**
+     * Create a new definition.
+     *
+     * @param n a node
+     */
+    public ReachingDefinitionsValue(AssignmentNode n) {
+        this.def = n;
+    }
+
+    @Override
+    public int hashCode() {
+        return this.def.hashCode();
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
+        if (!(obj instanceof ReachingDefinitionsValue)) {
+            return false;
+        }
+        ReachingDefinitionsValue other = (ReachingDefinitionsValue) obj;
+        return this.def.equals(other.def);
+    }
+
+    @Override
+    public String toString() {
+        return this.def.toString();
```

Comment:
Adding some prefix would be useful, no?

Reply:
Do you mean I can write this as `return ReachingDefinitionsValue: + this.def.toString();`, just like your previous comment?

Reply:
Well, if I add prefix here, the output will look weird.
From `reaching definitions = a = 1, b = 2, c = 3, e = (a + b), d = (a + c)` to `reaching definitions = Prefix: a = 1, Prefix: b = 2, Prefix: c = 3, Prefix: e = (a + b), Prefix: d = (a + c)`